xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -117,6 +117,7 @@ SelectionStream& CSelectionStreams::Get(StreamType type, int index)
 
 std::vector<SelectionStream> CSelectionStreams::Get(StreamType type)
 {
+  CSingleLock lock(m_section);
```

Comment:
VideoPlayer has its own thread and only this thread is supposed to access this method. It was designed only for use in OpenDefaultStreams: https://github.com/xbmc/xbmc/pull/700/commits/240692b2a3a4b9675bb4c972e745d7a1e8e1c3de

Unfortunately folks started to use it for different purpose like exposing information to the GUI. When calling this method the vector is copied twice which is a performance issue.